app/models/user.rb

```diff
@@ -1,36 +1,66 @@
 class User < ApplicationRecord
+  attr_accessor :remember_token
   before_save :downcase_email
   has_secure_password
   has_secure_password :recovery_password, validations: false
 
   validates :name,
             presence: true,
-            length: {minimum: 4}
             length: {minimum: Settings.length.digit_4}
 
   validates :account_name,
             presence: true,
             uniqueness: true,
-            length: {in: 6..20}
             length: {in: Settings.length.digit_6..Settings.length.digit_20}
 
   validates :email,
             presence: true,
             uniqueness: true,
-            length: {maximum: 150},
             length: {maximum: Settings.length.digit_150},
             format: {with: URI::MailTo::EMAIL_REGEXP}
 
   validates :age,
             presence: true,
-            length: {in: 1..3}
             length: {in: Settings.length.digit_1..Settings.length.digit_3}
 
   validates :gender,
             presence: true,
             inclusion: {in: %w(male female other)}
 
+  validates :password,
+            presence: true,
+            length: {minimum: Settings.length.digit_6},
+            allow_nil: true
+
   def downcase_email
     email.downcase!
   end
+
+  def self.digest string
+    cost = if ActiveModel::SecurePassword.min_cost
+             BCrypt::Engine::MIN_COST
+           else
+             BCrypt::Engine.cost
+           end
+    BCrypt::Password.create string, cost: cost
+  end
```

Comment:
đưa vào trong block class << self ở dưới luôn nhé
```suggestion
class << self
  def digest string
    cost = if ActiveModel::SecurePassword.min_cost
             BCrypt::Engine::MIN_COST
           else
             BCrypt::Engine.cost
           end
    BCrypt::Password.create string, cost: cost
  end
  
  def new_token
  ...
```